spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/StreamCommands.java

```diff
@@ -116,9 +121,19 @@ public String deployStream(
 				propertiesToUse = DeploymentPropertiesUtils.parse(properties);
 				break;
 			case 1:
-				Properties props = new Properties();
-				try (FileInputStream fis = new FileInputStream(propertiesFile)) {
-					props.load(fis);
+				String extension = FilenameUtils.getExtension(propertiesFile.getName());
```

Comment:
It may (or may not) be worth having a look at the `PropertySourceLoader` class and friends in boot